Private events to development branch, project private members

SamroodAli opened the pull request:
I was asked to build a project which allows members to create events and others can attend it

- [x] Users can create, attend events

- [x] Put navigation links across the top to help you jump around.

- [x] make an Event’s Show page display a list of attendees.

- [x] association between the event attendee (also a User) and the event. Call this user the “attendee”. Call the event the “attended_event”. You’ll again need to juggle specially named foreign keys and classes and sources.

- [x] I have used scope for the event methods [past and upcoming]

- [x] Add the association between the event creator (a User) and the event. Call this user the “creator”. Add the foreign key to the Events model as necessary. You’ll need to specify your association properties carefully (e.g. :foreign_key, :class_name, and :source).

MahmoudBakr23 commented on app/controllers/events_controller.rb:
```diff
@@ -0,0 +1,35 @@
+class EventsController < ApplicationController
+  def index
+    if signed_in?
+      @my_events = Event.where(creator_id: current_user.id)
+      @other_events = Event.where.not(creator_id: current_user.id)
+    else
+      redirect_to root_path
+    end
+  end
```
Please separate the events to upcoming and past events the way you did it on every single user's profile page.
```
@past_events = Event.previous_events
@upcoming_events = Event.upcoming_events
```

MahmoudBakr23 commented on config/routes.rb:
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  devise_for :users, controllers: { sessions: 'users/sessions' ,registrations: 'users/registrations'}
+  resources :events
+  resources :tickets
```
```
resources :events, only: %i[index new create show]
resources :tickets, only: %i[create]
```

MahmoudBakr23 reviewed (changes requested):
## **Changes Required 🛑♻**

<hr>

Hello there :)
You've made a great app right there. keep up the great work and please consider my comments below 👍 

### **Highlights**

- Linter is well-configured 💯
- Descriptive PR summary and commit messages 💯
- Correct models' associations and database connection 💯
- Extra credit for the cool design 💯
- Extra credit for making it possible for the users to update their accounts 💯

However, there is always room for improvement 👍 

### **To Improve**

- [ ] Please revise your migrations and remove what causes the error in the image down below.
![Git-Bksh__c_Users_mbakr_OneDrive_Desktop_scaffold_scaff_private-events 5_14_2021 6_12_54 PM](https://user-images.githubusercontent.com/47625407/118302069-1272f900-b4e4-11eb-9dc0-35a49658a268.png)

- [ ] Kindly modify the Event's controller index method to justify the project specifications as shown below and in my inline comments.
![past_upcoming_events_requirement](https://user-images.githubusercontent.com/47625407/118302204-38989900-b4e4-11eb-972f-fb2d8a339269.png)

- [ ] Please make sure that a user can only attend an event once. Take a look at the clarifying image below.
![Screenshot](https://user-images.githubusercontent.com/47625407/118302323-5a921b80-b4e4-11eb-8ae4-63182bf82e18.png)

- [ ] Please adjust the routes of the events' resources to only fit the needs. Kindly consider my inline comments on this point.
- [ ] Kindly keep in mind that you should test the app's units and integrations. Currently, there are 16 examples, 4 failures, and 12 pending. Please work on that.
![Git-Bksh__c_Users_mbakr_OneDrive_Desktop_scaffold_scaff_private-events 5_14_2021 6_34_47 PM](https://user-images.githubusercontent.com/47625407/118302757-e86e0680-b4e4-11eb-9804-9acf7104a883.png)

- [ ] In your README file, please add the CLI ```rails webpacker:install``` because you're using it in the app.
- [ ] In your README file, please add the CLI ```rails db:create``` before the one ```rails db:migrate```


### **OPTIONAL >> Nice to have**

- You should know that the views are only for rendering the app's visual items only and not for any logic. So, it'd be great if you move the logic from the views to the helpers.
- It'd be nice if you included a screenshot of the app in the README file.

<hr>

Kindly do the required changes and ask for another CRR.
Good luck and happy coding. 👨‍💻💻
If you got any questions, please feel free to contact me on Slack with the name "Mahmoud Bakr".

SamroodAli commented:
Please review my code again.
- [ ] I have fixed migration issue

SamroodAli commented:
Please review my code

- [ ] I have fixed the migrations issue

- [ ]  modified controller index method to past_upcoming_events_requirement

- [ ] I have added Event.past and Event.upcoming in event controller for event index page

- [ ]  Event can only be attended once, repeated attendance have been fixed
- [ ] Events routes have been reduced to only fit the needs
- [ ] Tests for models and integration tests are written
- [ ] In readme, webpacker:install requirement have been mentioned
- [ ] In readme, db:create and db:migrate have also been included

Optional nice to have recommended

- [ ] Numerous helpers have been introduced in views to remove logic from views
- [ ] Screenshot of app has been added

Additionally
- [ ] Option to cancel attending event has been added
- [ ] Attend and Cancel only for upcoming events

mimipeshy commented:
## STATUS: CHANGES REQUESTED 🔴 ♻️ 

Hello there @SamroodAli 👋🏾 👋🏾 great job on this milestone. You have worked on the comments left by the previous TSE, keep it up 😄 . 
However, before marking the project as complete please fix the following error

- [ ] One of your tests is failing, kindly fix that ⬇️ 

![image](https://user-images.githubusercontent.com/33067256/118486895-ceba0280-b722-11eb-833b-caf67b967603.png)

That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

SamroodAli commented:
All the tests are passing. I have run it many times now. Please let me know if the issue persists.

andresporras3423 reviewed (approved):
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:
**Hello Sam Ali** 
Amazing job :tada: :tada: :tada:  you have met all the requirements for this project
Now you are ready to continue with your next challenge :muscle: 
Don't forget to merge with your develop branch
 
Keep going like this  :clap: :clap:
 
Oscar Andrés Russi Porras
